src/navigator-card-body.jsx

```diff
@@ -0,0 +1,302 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useCallback, useEffect, useMemo, useState, useRef } from "react";
+import {
+    Card, CardBody,
+    Flex,
+    Gallery,
+    Icon,
+    CardTitle, Spinner, CardHeader,
+} from "@patternfly/react-core";
+import { FileIcon, FolderIcon } from "@patternfly/react-icons";
+
+import cockpit from "cockpit";
+import { useDialogs } from "dialogs.jsx";
+import { ListingTable } from "cockpit-components-table.jsx";
+
+const _ = cockpit.gettext;
+
+const compare = (sortBy) => {
+    const compareFileType = (a, b) => {
+        const aIsDir = (a.type === "dir" || a?.to === "dir");
+        const bIsDir = (b.type === "dir" || b?.to === "dir");
+
+        if (aIsDir && !bIsDir)
+            return -1;
+        if (!aIsDir && bIsDir)
+            return 1;
+        return 0;
+    };
+
+    switch (sortBy) {
+    case "az":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.name.toLowerCase() < b.name.toLowerCase()
+                ? -1
+                : 1)
+            : compareFileType(a, b);
+    case "za":
+        return (a, b) => compareFileType(a, b) === 0
+            ? (a.name.toLowerCase() > b.name.toLowerCase()
+                ? -1
+                : 1)
+            : compareFileType(a, b);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-231-20240130-170704-96686a3f-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)